torch/onnx/ops/_impl.py

```diff
@@ -56,18 +56,54 @@ def rotary_embedding_23(
     rotary_embedding_dim: int = 0,
 ) -> torch.Tensor:
     """RotaryEmbedding-23 https://onnx.ai/onnx/operators/onnx__RotaryEmbedding.html#rotaryembedding-23"""
+    # x has shape (batch_size, num_heads, sequence_length, head_size)
+    # or (batch_size, sequence_length, hidden_size)
+    input_shape = x.shape
+    input_rank = len(input_shape)
+    batch_size = input_shape[0]
+    sequence_length = input_shape[-2]
```

Comment:
If 4d, this is input_shape[-1]

Reply:
I though sequence_length is always [-2]? When 4d it is (batch_size, num_heads, sequence_length, head_size)